superset/views/core.py

```diff
@@ -298,6 +304,129 @@ class DatabaseAsync(DatabaseView):
 appbuilder.add_view_no_menu(DatabaseAsync)
 
 
+@app.route('/uploads/<filename>')
+def uploaded_file(filename):
+    with open("flask-stream-demo", "bw") as f:
+        chunk_size = 4096
+        while True:
+            chunk = flask.request.stream.read(chunk_size).decode('utf-8')
+            if len(chunk) == 0:
+                return
+            f.write(chunk)
+
+    #return send_from_directory(config['UPLOAD_FOLDER'],
+    #                           filename)
+
+class CsvToDatabaseView(SimpleFormView):
+    #DatasourceModelView,  add to ^ ?
+    form = CsvToDatabaseForm
+    form_title = _('CSV to Database configuration')
+    add_columns = ['database', 'schema', 'table_name']
+
+    def form_get(self, form):
+        # pre process form
+        # default values
+        form.name.data = None
+        form.csv_file.data = None
+        form.sep.data = ','
+        form.header.data = 0
+        form.index_col.data = None
+        form.squeeze.data = False
+        form.prefix.data = None
+        form.mangle_dupe_cols.data = True
+        form.skipinitialspace.data = False
+        form.skiprows.data = None
+        form.nrows.data = None
+        form.skip_blank_lines.data = True
+        form.parse_dates.data = True
+        form.infer_datetime_format.data = True
+        form.dayfirst.data = False
+        form.thousands.data = None
+        form.decimal.data = '.'
+        form.quotechar.data = None
+        form.escapechar.data = None
+        form.comment.data = None
+        form.error_bad_lines.data = False
+        form.names.data = None
+        form.con.data = config['SQLALCHEMY_DATABASE_URI']
+        form.schema.data = None
+        form.if_exists.data = 'append'
+        form.index.data = None
+        form.index_label.data = None
+
+    def form_post(self, form):
+        # post process form
+
+        # Turn into list of strings
+        def upload_file(csv_file):
+            if csv_file and csv_file.filename:
+                filename = secure_filename(csv_file.filename)
+                csv_file.save(os.path.join(config['UPLOAD_FOLDER'],
+                                                 filename))
+                return filename
+
+
+        if form.names.data is not None:
+            form.names.data = form.names.data.split(",")
+        else:
+            if form.header.data is None:
+                form.header.data = 0
+
+
+        datasources = ConnectorRegistry.get_all_datasources(db.session)
+        filename = upload_file(form.csv_file.data)
+
+        # some of this stuff belongs in db_engine_spec
+        #print(form.data)
```

Comment:
please clean up those artifacts, makes me feel like this PR isn't ready for review